### PR TITLE
Adds guide for setting up mobile SDKs proxy

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do? What is the motivation?
Adds documentation guide on how to setup a proxy for mobile SDKs

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+When initializing the Android SDK, specify the following proxy configuration:
+
+```kotlin
+val configBuilder = Configuration.Builder(
+    clientToken = "<client token>",
+    env = "<environment>"
+)
+
+val proxy = Proxy(Proxy.Type.HTTP, InetSocketAddress("www.example.com", 123))
+val authenticator = ProxyAuthenticator("proxyuser", "proxypass")
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:38]:
```suggestion
val proxy = Proxy(Proxy.Type.HTTP, InetSocketAddress("<www.example.com>", <123>))
val authenticator = ProxyAuthenticator("<proxy user>", "<proxy password>")
```

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+{{% tab "React Native" %}}
+
+When initializing the React Native SDK, specify the following proxy configuration:
+
+```javascript
+import { DatadogProviderConfiguration, ProxyConfiguration, ProxyType } from '@datadog/mobile-react-native';
+
+const config = new DatadogProviderConfiguration('<client token>', '<environment>', '<application id>');
+
+config.proxyConfig = new ProxyConfiguration(ProxyType.HTTPS, 'www.example.com', 123, 'proxyuser', 'proxypass');
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:85]:
```suggestion
config.proxyConfig = new ProxyConfiguration(ProxyType.HTTPS, '<www.example.com>', <123>, '<proxy user>', '<proxy password>');
```

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+When initializing the Android SDK, specify the following proxy configuration:
+
+```kotlin
+val configBuilder = Configuration.Builder(
+    clientToken = "<client token>",
+    env = "<environment>"
+)
+
+val proxy = Proxy(Proxy.Type.SOCKS, InetSocketAddress("www.example.com", 123))
+val authenticator = ProxyAuthenticator("proxyuser", "proxypass")
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:110]:
```suggestion
val proxy = Proxy(Proxy.Type.HTTP, InetSocketAddress("<www.example.com>", <123>))
val authenticator = ProxyAuthenticator("<proxy user>", "<proxy password>")
```

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+Datadog.initialize(
+  with: Datadog.Configuration(
+    clientToken: "<client token>",
+    env: "<environment>",
+    proxyConfiguration: [
+        kCFNetworkProxiesSOCKSEnable: true,
+        kCFNetworkProxiesSOCKSPort: 123,
+        kCFNetworkProxiesSOCKSProxy: "www.example.com",
+        kCFProxyUsernameKey: "proxyuser",
+        kCFProxyPasswordKey: "proxypass"
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:135]:
```suggestion
        kCFNetworkProxiesSOCKSPort: <123>,
        kCFNetworkProxiesSOCKSProxy: "<www.example.com>",
        kCFProxyUsernameKey: "<proxy user>",
        kCFProxyPasswordKey: "<proxy password>"
```

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+{{% tab "React Native" %}}
+
+When initializing the React Native SDK, specify the following proxy configuration:
+
+```javascript
+import { DatadogProviderConfiguration, ProxyConfiguration, ProxyType } from '@datadog/mobile-react-native';
+
+const config = new DatadogProviderConfiguration('<client token>', '<environment>', '<application id>');
+
+config.proxyConfig = new ProxyConfiguration(ProxyType.SOCKS, 'www.example.com', 123, 'proxyuser', 'proxypass');
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:156]:
```suggestion
config.proxyConfig = new ProxyConfiguration(ProxyType.SOCKS, '<www.example.com>', <123>, '<proxy user>', '<proxy password>');
```

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+Datadog.initialize(
+  with: Datadog.Configuration(
+    clientToken: "<client token>",
+    env: "<environment>",
+    proxyConfiguration: [
+        kCFNetworkProxiesHTTPEnable: true,
+        kCFNetworkProxiesHTTPPort: 123,
+        kCFNetworkProxiesHTTPProxy: "www.example.com",
+        kCFProxyUsernameKey: "proxyuser",
+        kCFProxyPasswordKey: "proxypass"
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:64]:
```suggestion
        kCFNetworkProxiesHTTPPort: <123>,
        kCFNetworkProxiesHTTPProxy: "<www.example.com>",
        kCFProxyUsernameKey: "<proxy user>",
        kCFProxyPasswordKey: "<proxy password>"
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Looks good overall, just requesting some minor updates. 

Additionally, could you please remove the changes to the French language documentation? These changes would ultimately be overwritten in the build process, because translated documentation is updated by a separate service.

Let me know if you have any questions!

**Review by 0xnm — APPROVED** [review]

**Comment by plousada** [comment]
In this same PR we should also remove the reference to proxies contained in iOS's (and I think RN too) advanced configuration RUM configuration doc..

**Review by BenoitZugmeyer — APPROVED** [review]
LGTM

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md [hunk 7]
```diff
@@ -58,16 +58,32 @@ Datadog.initialize(
...
+@import DatadogObjc;
+
+DDConfiguration *configuration = [[DDConfiguration alloc] initWithClientToken:@"<client token>" env:@"<environment>"];
+configuration.proxyConfiguration = @{
+    (NSString *)kCFNetworkProxiesHTTPEnable: @YES,
+    (NSString *)kCFNetworkProxiesHTTPPort: @<123>,
+    (NSString *)kCFNetworkProxiesHTTPProxy: @"<www.example.com>",
+    (NSString *)kCFProxyUsernameKey: @"<proxyuser>",
+    (NSString *)kCFProxyPasswordKey: @"<proxypass>"
+}];
```
Comment (buraizu) [inline content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:81]: Just wanted to flag that it looks like there's no opening bracket (`[`) for the closing bracket on line 81. Feel free to ignore this comment if I'm missing anything.

**Review by buraizu — APPROVED** [review]
Thanks for applying those updates. I've just flagged one potential issue in the new content, but overall LGTM and approving. Please let me know if/when this PR is ready to be merged.

Also, could you please remove the updates to the French language documentation? While appreciated, these changes would ultimately be overwritten, since changes to the translated documentation are handled through a separate process. Let me know if you have any questions!

File: content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md (hunk 7 above)
Comment (louiszawadzki) [reply content/en/real_user_monitoring/guide/proxy-mobile-rum-data.md:81]: Good catch it was a mistake I carried over from the previous doc, I will fix it!

**Comment by louiszawadzki** [comment]
@buraizu I've made the final edits :) 

Since it's been approved by the mobile SDK team can you merge it?
Thanks a lot!

**Pull request merged**